src/classes/UTIL_SortContacts.cls

```diff
@@ -0,0 +1,105 @@
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Utilities
+* @description Utility for Sorting Contacts
+*/
+public class UTIL_SortContacts implements Comparable {
+    public String FirstName;
+    public String LastName;
+    public Contact contactRecord;
+
+    /*******************************************************************************************************
+    * @description Constructor that sets the Contact's First and Last Name and the Contact record itself.
+    * @param c is the Contact record
+    */
+    public UTIL_SortContacts(Contact c) {
+        FirstName = c.FirstName;
+        LastName = c.LastName;
+        contactRecord = c;
+    }
+
+    /*******************************************************************************************************
+    * @description method that returns an Integer value based on LastName/FirstName comparison.
+    * @param compareTo is the object that is being passed for comparison.
+    * @return Integer
+    */
+    public Integer compareTo(Object compareTo) {
+        return sortByLastNameFirstName(compareTo);
+    }
+
+    /*******************************************************************************************************
+    * @description sorts Contact by LastName and FirstName.
+    * @param compareTo is the object that is being passed for comparison.
+    * @return Integer
+    */
+    private Integer sortByLastNameFirstName(Object compareTo) {
+        UTIL_SortContacts contactToCompare = (UTIL_SortContacts)compareTo;
+
+        if (this.LastName == contactToCompare.LastName) {
+            return sortByFirstName(contactToCompare);
+        } else {
+            return sortByLastName(contactToCompare);
+        }
+
+    }
+
+    /*******************************************************************************************************
+    * @description sorts Contact by FirstName.
+    * @param contactToCompare is the record that needs to be compared
+    * @return Integer
+    */
+    private Integer sortByFirstName(UTIL_SortContacts contactToCompare) {
+        if (this.FirstName == contactToCompare.FirstName) {
+            return 0;
```

Comment:
This whole class is a thing of beauty, and nicely formatted too!
I was considering asking if we wanted to make these methods "TestVisible" but I think we are okay since in order to test these methods individually we'd need to setup the class anyway.

The only other thing I'd ask is the return statements are 1, -1, and 0 -- should we consider CONSTANTS to make those outcomes more readable? At first glance, I don't know if 1 means I should come before or after what I'm being compared to, etc. Just a thought but nothing I'd hold up merging for.

Reply:
I've added the 3 constants. I struggled a little on how to name 0. I just named it "SORT_SAME". O_O Idk if that makes sense. I'm open to suggestions.